topdown/strings.go

```diff
@@ -411,28 +412,50 @@ func builtinTrimSpace(a ast.Value) (ast.Value, error) {
 
 func builtinSprintf(a, b ast.Value) (ast.Value, error) {
 	s, err := builtins.StringOperand(a, 1)
+
 	if err != nil {
 		return nil, err
 	}
 
 	astArr, ok := b.(*ast.Array)
+
 	if !ok {
 		return nil, builtins.NewOperandTypeErr(2, b, "array")
 	}
 
 	args := make([]interface{}, astArr.Len())
 
+	var formatStringArr []string
+	if len(args) > 0 {
+		// Split up the format string
+		regex, err := regexp.Compile(`%(((\.[0-9]+)|([0-9]+)|([0-9]+\.[0-9]+))([a-zA-Z])|([a-zA-Z]))`)
+
+		formatStringArr = regex.FindAllString(s.String(), -1)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(formatStringArr) != len(args) {
+		// Throw exception - not even args for sprintf function
+		return nil, builtins.NewOperandErr(2, "There are not enough arguments for sprintf function. Please advise.")
```

Comment:
I this is the case we're hitting when args are explicitly indexed?

👉 Here's what the stdlib does, perhaps it could serve as inspiration? https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/fmt/print.go;l=974-1125;drc=e7c56fe9948449a3710b36c22c02d57c215d1c10

Reply:
Yes, in the event that we have an argument that does not have enough arguments, then we're going to throw an exception.

Reply:
With the explicit argument indexes, I'll need to modify this to accommodate it.